providers/linux/util.go

```diff
@@ -26,20 +26,27 @@ import (
 	"strconv"
 )
 
-func parseKeyValue(content []byte, separator string, callback func(key, value []byte) error) error {
-	sc := bufio.NewScanner(bytes.NewReader(content))
-	for sc.Scan() {
-		parts := bytes.SplitN(sc.Bytes(), []byte(separator), 2)
-		if len(parts) != 2 {
+// parseKeyValue parses KEY<separator>VALUE from each line in content and invokes callback(KEY, VALUE).
+// White-space is trimmed from VALUE. Empty lines are ignored. All non-empty lines must contain
+// the separator otherwise an error is returned.
```

Comment:
More idiomatic doc would be something like (not required for approval)
```suggestion
// parseKeyValue parses key/val pairs separated by the provided separator from
// each line in content and invokes the callback. White-space is trimmed from
// val. Empty lines are ignored. All non-empty lines must contain the separator
// otherwise an error is returned.
```